sqlmesh/core/engine_adapter/base.py

```diff
@@ -53,6 +53,7 @@
         PySparkSession,
         Query,
         QueryOrDF,
+        SnowparkSession,
```

Comment:
I don't understand how this import can possible work with Python 3.12

Reply:
because it's an if else statement in the typing

Reply:
ah i see, you're right, i don't know how it works